packages/odata-service-inquirer/src/prompts/datasources/sap-system/system-selection/questions.ts

```diff
@@ -0,0 +1,259 @@
+        if (isPartialUrlDestination(systemSelection.system as Destination)) {
+            // Reset the connection state as we are deferring the connection validation to the service path prompt, any existing connection state should be cleared
+            // connectionValidator.resetConnectionState(true);
+            return true;
+        }
+        connectValResult = await connectWithDestination(
+            systemSelection.system as Destination,
+            connectionValidator,
+            requiredOdataVersion
+        );
+    }
+    return connectValResult;
+}
+
+/**
+ * Returns a list of questions for creating a new system configuration or selecting an existing stored system.
+ *
+ * @param promptOptions prompt options that may be used to customize the questions
+ * @returns a list of questions for creating a new system configuration or selecting an existing stored system
+ */
+export async function getSystemSelectionQuestions(
+    promptOptions?: OdataServicePromptOptions
+): Promise<Question<SystemSelectionAnswers & ServiceAnswer>[]> {
+    PromptState.reset();
+    const connectValidator = new ConnectionValidator();
+
+    const questions: Question<SystemSelectionAnswers & ServiceAnswer>[] = await getSystemConnectionQuestions(
+        connectValidator,
+        promptOptions
+    );
+
+    // Existing system (BackendSystem or Destination) selected,
+    // In future, make the service prompt optional by wrapping in condition `[promptOptions?.serviceSelection?.hide]`
+    questions.push(
+        ...withCondition(
+            getSystemServiceQuestion(connectValidator, systemSelectionPromptNamespace, promptOptions) as Question[],
+            (answers: Answers) => (answers as SystemSelectionAnswers).systemSelection?.type !== 'newSystemChoice'
+        )
+    );
+
+    // Create new system connection for storage only supported on non-App Studio environments
+    if (!isAppStudio()) {
+        questions.push(
+            ...withCondition(
+                getNewSystemQuestions(promptOptions) as Question[],
+                (answers: Answers) => (answers as SystemSelectionAnswers).systemSelection?.type === 'newSystemChoice'
+            )
+        );
+    }
+
+    return questions;
+}
+
+/**
+ * Returns a list of existing systems, either destinations or backend systems from persistent store, depending on the environment.
+ * Note that destinations are only available in BAS environment and must include the destination attribute `WebIDEEnabled` to be listed.
+ * Additional destination attribute filters may be provided.
+ *
+ * @param connectionValidator A reference to the active connection validator, used to validate the service selection and retrieve service details.
+ * @param promptOptions
+ * @returns a list of existing systems
+ */
+export async function getSystemConnectionQuestions(
+    connectionValidator: ConnectionValidator,
+    promptOptions?: OdataServicePromptOptions
```

Comment:
No, because its sharing a prompt option with service selection ( to narrow the catalog requests to a specific odata version). I didnt want 2 options on different prompts that do the same thing but could conflict.

Reply:
Thanks for the explanation

Reply:
Ok thanks for the explanation